daemon/exec.go

```diff
@@ -276,15 +277,11 @@ func (daemon *Daemon) ContainerExecStart(ctx context.Context, name string, stdin
 	select {
 	case <-ctx.Done():
 		logrus.Debugf("Sending TERM signal to process %v in container %v", name, c.ID)
-		daemon.containerd.SignalProcess(ctx, c.ID, name, int(signal.SignalMap["TERM"]))
-
-		timeout := time.NewTimer(termProcessTimeout)
-		defer timeout.Stop()
-
+		daemon.ContainerExecKill(ctx, name, uint64(signal.SignalMap["TERM"]))
 		select {
-		case <-timeout.C:
-			logrus.Infof("Container %v, process %v failed to exit within %v of signal TERM - using the force", c.ID, name, termProcessTimeout)
-			daemon.containerd.SignalProcess(ctx, c.ID, name, int(signal.SignalMap["KILL"]))
+		case <-time.After(termProcessTimeout):
```

Comment:
Any reason this was changed from NewTimer + defer stop?

Note that time.After holds resources until it fires even if this select statement is already completed by another case, like attachErr, unlike the defer Stop case which frees it explicitly.

https://golang.org/pkg/time/#After

Reply:
Sorry, I'm not familiar with Go (this carry of the PR was my first experience), would you have a moment to give me a snippet of code on what should I change exactly?

I'd gladly do the change and monitor the CI.

Reply:
Ah no worries mate, I actually just meant that this particular part of the diff here can just be reverted, the old timer code before this PR was more efficient:

```go
timeout := time.NewTimer(termProcessTimeout)	
defer timeout.Stop()

select {
case <-timeout.C:
```

That code was replaced by `case <-time.After(termProcessTimeout)`, which while shorter, is less efficient at managing resources in this situation.

Reply:
Ok, thanks for the clarification! I've just pushed a commit that (I think) it effectively reverts to the previous logic.